apps/rush-lib/src/logic/taskRunner/ProjectTask.ts

```diff
@@ -31,13 +31,11 @@ export interface IProjectTaskOptions {
 function _areShallowEqual(object1: JsonObject, object2: JsonObject, writer: ITaskWriter): boolean {
   for (const n in object1) {
     if (!(n in object2) || object1[n] !== object2[n]) {
-      writer.writeLine(`Found mismatch: "${n}": "${object1[n]}" !== "${object2[n]}"`);
       return false;
     }
   }
   for (const n in object2) {
     if (!(n in object1)) {
-      writer.writeLine(`Found new prop in obj2: "${n}" value="${object2[n]}"`);
```

Comment:
Change these to print only in debug mode rather than removing these completely?

Reply:
That is a good idea!

Reply:
What is "debug mode"?

Reply:
BTW my interpretation is that this `_areShallowEqual()` is simply a utility for copying two JSON objects, which returns `true` or `false`.  These messages are printed when it returns `false` but that doesn't indicate any problem, any more than inappropriately returning `true` could be a problem.  Thus these messages are really only useful to a person who is debugging Rush.

If we don't want to delete them entirely (which I was fine with), maybe they could just be code comments. (?)

Reply:
I was thinking the debug mode Sachin refers might be something like 
```
@debug
writer.writeLine(`Found new prop in obj2: "${n}" value="${object2[n]}"`);
```
Java has such feature and I'm looking into if Typescript has it or not. But I haven't found any.  @sachinjoseph Any suggestion here? Or we can just comment them as a reference for debugging :)

Reply:
That isn't a feature in TypeScript.

Let's just comment them out.

Reply:
What I meant was "rush --debug build"

Reply:
We really should provide a way for Rush to produce diagnostic logging output and optionally redirect it to a file.  This "logging" should be a separate channel from the "console" output that interacts with users.  (The concepts are fundamentally different, e.g. logging has verbosity levels and sometimes categories, whereas console output has colors and cursor movements.)

As far as I know, it's out of scope for this PR, though.  We should open a separate GitHub issue.